src/commands/git-metadata/gitdb.ts

```diff
@@ -116,13 +118,25 @@ const getLatestLocalCommits = async (git: simpleGit.SimpleGit) => {
 }
 
 const unshallowRepositoryWhenNeeded = async (log: Logger, git: simpleGit.SimpleGit) => {
-  const isShallow = (await git.revparse('--is-shallow-repository')) === 'true'
-  if (isShallow) {
+  const shallowCmdPromise = git.revparse('--is-shallow-repository')
+  const versionCmdPromise: simpleGit.Response<simpleGit.VersionResult> = git.version()
+  const isShallow = (await shallowCmdPromise) === 'true'
+  const gitversion = String(await versionCmdPromise)
+  if (isShallow && gte(gitversion, '2.27.0')) {
     log.info('[unshallow] Git repository is a shallow clone, unshallowing it...')
```

Comment:
nitpick/idea: how would this look?
```suggestion
  const isShallow = (await git.revparse('--is-shallow-repository')) === 'true'
  if (!isShallow) {
    return
  }
  const gitversion = String(await git.version())
  if (lte(gitversion, '2.27.0')) {
    return
  }
  log.info('[unshallow] Git repository is a shallow clone, unshallowing it...')
```

the early returns are a matter of taste, so feel free to keep the logic as is. We could, however: 
* get `isShallow` and `gitversion` directly, without passing through `shallowCmdPromise` and `versionCmdPromise`
* avoid calling `git.version()` if the repository isn't shallow